File: 잡동사니/k2022/spring/연습/voucher/src/main/java/com/karma/voucher/model/AuditingFields.java
package com.karma.voucher.model;

import org.springframework.data.annotation.CreatedDate;
import org.springframework.data.annotation.LastModifiedBy;
import org.springframework.data.jpa.domain.support.AuditingEntityListener;

import javax.persistence.Column;
import javax.persistence.EntityListeners;
import javax.persistence.MappedSuperclass;
import java.time.LocalDateTime;

@MappedSuperclass
@EntityListeners(AuditingEntityListener.class)
public abstract class AuditingFields {
    @CreatedDate @Column(updatable = false,nullable = false)
    private LocalDateTime createdAt;

    @LastModifiedBy @Column
    private LocalDateTime modifiedAt;
}

File: backend/backend/src/main/kotlin/com/karma/chat/domain/auditing/AuditingFields.kt
package com.karma.chat.domain.auditing

import jakarta.persistence.*
import org.springframework.data.annotation.CreatedBy
import org.springframework.data.annotation.CreatedDate
import org.springframework.data.annotation.LastModifiedBy
import org.springframework.data.annotation.LastModifiedDate
import org.springframework.data.jpa.domain.support.AuditingEntityListener
import org.springframework.format.annotation.DateTimeFormat
import java.time.LocalDateTime

@EntityListeners(AuditingEntityListener::class)
@MappedSuperclass
abstract class AuditingFields {
    @Id @GeneratedValue(strategy = GenerationType.AUTO)
    @Column(name="id")
    open var id: Long = 0
        protected set

    @DateTimeFormat(iso = DateTimeFormat.ISO.DATE_TIME)
    @CreatedDate
    @Column(updatable = false)
    open var createdAt: LocalDateTime? = null
        protected set

    @CreatedBy
    @Column(updatable = false, length = 100)
    open var createdBy: String? = null
        protected set

    @DateTimeFormat(iso = DateTimeFormat.ISO.DATE_TIME)
    @LastModifiedDate
    open var modifiedAt: LocalDateTime? = null
        protected set

    @LastModifiedBy
    @Column(length = 100)
    open var modifiedBy: String? = null
        protected set
}